frontend/src/core/comments/components/AddComment.js

```diff
@@ -1,97 +1,393 @@
+        }
 
-    const handleOnChange = () => {
-        const textAreaLineHeight = 24;
-        commentInput.current.rows = minRows;
+        // If suggestions in team comments scroll below or suggestions in translation
+        // comments scroll above the next section or overflow the window then hide the suggestions
+        if ((teamCommentsRect && teamCommentsActive && 
+                (((setTop + suggestionsHeight) - editorMenuHeight) > teamCommentsRect.height)) ||
+            (translateCommentsRect && translateCommentsActive && (rect.top < translateCommentsRect.top)) || 
+            (translateCommentsRect && translateCommentsActive && (setTop + suggestionsHeight > window.innerHeight))) {
+            el.style.display = 'none';
+        }
+        
+        // If suggestions overflow the window width in team comments or the right side of the 
+        // translations comments then adjust the position so they display to the left of the mention
+        const suggestionsWidth = el.clientWidth
+        const translateCommentsOverflow = !translateCommentsRect ? false :
+            setLeft + suggestionsWidth > translateCommentsRect.right;
+        
+        if (setLeft + suggestionsWidth > window.innerWidth || 
+            (translateCommentsActive && translateCommentsOverflow)) {
+            setLeft = rect.right - suggestionsWidth;
+        }
 
-        const currentRows = Math.trunc(commentInput.current.scrollHeight / textAreaLineHeight);
+        el.style.top = `${ setTop }px`;
+        el.style.left = `${ setLeft }px`;
+    
+    }, [ chars.length, editor, index, search, target, scrollPosition ]);
 
-        if (currentRows < maxRows) {
-            commentInput.current.rows = currentRows;
+    React.useEffect(() => {
+        // Flow does not recognize the event listeners with 'SyntheticEvent`,  
+        // so I'm ignoring the errors Flow throws here.
+        // $FLOW_IGNORE
+        const handleScroll = (e: SyntheticEvent<HTMLElement>) => {
+            const element = e.currentTarget;
+            setScrollPosition(element.scrollTop);
+        }
+
+        const historyScroll = document.querySelector('#history-list');
+        const teamsScroll = document.querySelector('#react-tabs-3');
+
+        if (!historyScroll && !teamsScroll) {
+            return;
+        }
+        
+        if (historyScroll) {
+            historyScroll.addEventListener('scroll', handleScroll);
         }
-        else {
-            commentInput.current.rows = maxRows;
+        if (teamsScroll) {
+            teamsScroll.addEventListener('scroll', handleScroll);
+        }
+
+        return () => {
+            if (historyScroll) {
+                historyScroll.removeEventListener('scroll', handleScroll);
+            }
+            if (teamsScroll) {
+                teamsScroll.removeEventListener('scroll', handleScroll);
+            }
+        }
+    }, [])
+
+    const handleMentionsKeyDown = React.useCallback(
```

Comment:
I don't think this needs to be a callback either. It should be fine as a simple function. It's not too big of a deal to recreate the function on each render, as it depends on many things that are going to trigger a re-render anyway. Also, it's not a dependency of another hook, so it doesn't have to be "callback'ed".